### ISSUE TITLE
Recording with firefox bug and recording blob files bug

### ISSUE THREAD

**Issue by yanshekki** [streamproc/MediaStreamRecorder#94]
Hi there,

This is my testing site:
https://www.extratown.com/streaming

1, When I use firefox recording video mediaRecorder.start(15000) or more time that can't normal output blob data. Maybe 300kb - 500kb per one blob many and fast output.

2, When I use chrome to recording video mediaRecorder.start(15000) that will output blob file(s) about size 15000kb + 300kb "two files" for one record.

3, The record blob I change to "new File(recordedBlobs, n + '.webm', {mimeType: 'video/webm'});" that only can play in Linux firefox browser! Mac os chrome firefox and Linux chrome can't play the flies (I have no windows to test sor). Why can't play? I find problem may the files is not complete that say 15ms per one video in the control bar just play 1ms-10ms.

My english not good, forgive me.
Thank you!

**Comment by noahjerreel** [comment]
Hello @yanshekki , can you try changing the recorder type to StereoAudioRecorder? Try also reducing the video size on your recorder?

Are you trying to merge the blobs? If you're trying to stream the blobs, why not stream it directly every 15 secs, I mean, when ondataavailable is fires, stream it directly?

**Comment by yanshekki** [comment]
hi @noahjerreel Is StereoAudioRecorder just audio? I tried 1000ms 5000ms 10000ms 15000ms 30000ms with mediaRecorder.start() that output blob is not normal.

Using the same code when recording blob information:
chrome:
https://www.extratown.com/public/2016/06/21/%E8%9E%A2%E5%B9%95%E5%BF%AB%E7%85%A7-2016-06-21-%E4%B8%8B%E5%8D%883.22.07.png
firefox:
https://www.extratown.com/public/2016/06/21/%E8%9E%A2%E5%B9%95%E5%BF%AB%E7%85%A7-2016-06-21-%E4%B8%8B%E5%8D%883.25.14.png
<script type="text/javascript">
    var TRACKERS = [
    'ws://www.extratown.com:8080',
    'wss://www.extratown.com:8443',
    'udp://www.extratown.com:8080',
    'http://www.extratown.com:8080/announce'
];
    var gumVideo = document.querySelector('video#gum');
    var client = new WebTorrent();
    var delaymin = 30;
    var delaymax = 150;
    var torrents = [];
    var recordedBlobs = [];
    var sth;
    var cot;
    
    $(document).ready(function () {
    
        $("#streaming input").on("click", function () {

            $(this).select();

        });
        
    });

    function startRecording() {
        
        $("#stopbutton").css('display' , 'block');
        $("#startbutton").css('display' , 'none');
        mediaRecorder = new MediaStreamRecorder(window.stream);
        mediaRecorder.stream = stream;
        mediaRecorder.mimeType = 'video/webm';
        mediaRecorder.videoWidth = 1280;
        mediaRecorder.videoHeight = 720;
        mediaRecorder.onstop = stopRecording;
        mediaRecorder.bufferSize = 0;
        mediaRecorder.ondataavailable = function(blob) {
            console.log(blob);
            if (blob.size > 1000){
                recordedBlobs.push(blob);
                var d = new Date();
                var n = d.getTime();
                var file = new File(recordedBlobs, n + '.webm', {
                    mimeType: 'video/webm'
                });
                client.seed(file, {
                    announce: TRACKERS
                }, function (torrent) {
                    streamingtorrent(torrent);
                    torrents.push(torrent);
                });
                recordedBlobs = [];
            }
        };
        var runtimemin = delaymin * 1000;
        mediaRecorder.start(runtimemin / 2);
        var runtimemax = delaymax * 1000;
        setTimeout(function(){
            cot = setInterval(clearoldtorrent , runtimemin);
        }, runtimemax);
        
    }
    
    function stopRecording(event){
        
        $("#startbutton").css('display' , 'block');
        $("#stopbutton").css('display' , 'none');
        
    }
    
    function streamingtorrent(torrent) {
    
        var postcode = 'xasANIZQvwjr92xWl3jm';
        var postpath = 'fbc28f2f8490e235eff3cf161bfc05c4';
        
        $.post( "https://www.extratown.com/streaming/post" , {code: postcode , path: postpath , hash: torrent.infoHash});

    }
    
    function clearoldtorrent(){
        
        var torrent = torrents[0];
        client.remove(torrent , function(err){
            if (err){
                console.log(err);
            }else{
                torrents.shift();
            }
        });
        
    }

    navigator.mediaDevices.getUserMedia({
        audio: true
        , video: true
    }).then(handleSuccess);

    function handleSuccess(stream) {
        window.stream = stream;
        if (window.URL) {
            gumVideo.src = window.URL.createObjectURL(stream);
        } else {
            gumVideo.src = stream;
        }
    }
</script>

You can try the link(use not the same browser and os):
https://www.extratown.com/streaming

I tried merge the blobs, the problem same here.

I doing the streaming is recorded the video may 15ms per one video then using webtorrent p2p bt to other user to watch. So recorded one file will make a hash to share and start the next record video, the other user will download the videos and play non-stop.

**Comment by noahjerreel** [comment]
Yup @yanshekki , but you can set the recorder type to StereoAudioRecorded to force the mediastreamrecorder use the webaudio api. I'm currently using it on my end and was able to capture both video and audio

**Comment by yanshekki** [comment]
hi @noahjerreel , should you please let me know more details with StereoAudioRecorded recording video and audio at same time? thank you for your help.

maybe i need to use server to convert format like this:
https://github.com/muaz-khan/WebRTC-Experiment/tree/master/ffmpeg

**Comment by noahjerreel** [comment]
Hello @yanshekki , I would love to, but I think it's better if @muaz-khan will tell you. I can't fully grasp it yet. Hope you would understand.

FFMpeg will surely help you, but the only thing you will consider is it's size. But if you could make it work, compressing it, then that's good for you
